include/deal.II/lac/vector.templates.h

```diff
@@ -134,12 +135,62 @@ namespace internal
                            tbb::auto_partitioner());
       }
     else if (vec_size > 0)
-      functor(0,vec_size);
 #endif
+      functor(0,vec_size);
   }
 
 
-  // Define the functors necessary to use SIMD with TBB.
+  // Define the functors necessary to use SIMD with TBB. we also include the
+  // simple copy and set operations
+
+  template <typename Number>
+  struct Vector_set
+  {
+    Number *val;
```

Comment:
Agreed, I will change into 'dst' and 'value' (result of copy-pasting one entry and naming the other)